docs/determinism.md

```diff
@@ -0,0 +1,13 @@
+# Determinism (GPU)
+
+## Compilation
+
+XLA is generally developed such that the compilation process is deterministic by choosing appropriate algorithms and data structures.
+
+One important exception is autotuning, in particular of GEMM fusions: due to fluctuations in measurements different kernels can be picked as the fastest ones in different compilation runs. [Persisted autotuning](./persisted_autotuning) can be used as a workaround.
```

Comment:
I'd phrase it differently, starting with "XLA compilation is deterministic with autotuning map fixed using ..."

Reply:
Rephrased.